cookbook/configuration/database_session_storage.rst

```diff
@@ -1,19 +1,149 @@
+How to Store Session in the Database
+====================================
 
 The default session storage of Symfony2 writes the session information to
 file(s). Most medium to large websites use a database to store the session
 values instead of files, because databases are easier to use and scale in a
 multi-webserver environment.
 
-Symfony2 has a built-in solution for database session storage called
-:class:`Symfony\\Component\\HttpFoundation\\Session\\Storage\\Handler\\PdoSessionHandler`.
-To use it, you just need to change some parameters in ``config.yml`` (or the
-configuration format of your choice):
+Symfony2 has two built-in solutions for database session storage one uses doctrine
+:class:`Symfony\\Bridge\\Doctrine\\HttpFoundation\\HttpFoundation\\DbalSessionHandler` 
+and the other uses PDO :class:`Symfony\\Component\\HttpFoundation\\Session\\Storage\\Handler\\PdoSessionHandler`.
 
+Using Doctrine to Store the Session in the Database
+---------------------------------------------------
+
+To use The DBAL session storage, you need to register a new service and configure
+ Symfony's session handling to use it:
+
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/config.yml
+        framework:
+            session:
+                # ...
+                handler_id:     session.handler.dbal_handler
+
+        services:
+           
+            session.handler.dbal_handler:
+                class:     Symfony\Bridge\Doctrine\HttpFoundation\DbalSessionHandler
+                arguments: ["@doctrine.dbal.default_connection"]
+
+    .. code-block:: xml
+
+        <!-- app/config/config.xml -->
+        <framework:config>
+            <framework:session handler-id="session.handler.dbal_handler" cookie-lifetime="3600" auto-start="true"/>
+        </framework:config>
+
+        <services>
+            <service id="session.handler.dbal_handler" class="Symfony\Bridge\Doctrine\HttpFoundation\DbalSessionHandler">
+                <argument type="service" id="doctrine.dbal.default_connection" />                
+            </service>
+        </services>
+
+    .. code-block:: php
+
+        // app/config/config.php
+        use Symfony\Component\DependencyInjection\Definition;
+        use Symfony\Component\DependencyInjection\Reference;
+
+        $container->loadFromExtension('framework', array(
+            ...,
+            'session' => array(
+                // ...,
+                'handler_id' => 'session.handler.dbal_handler',
+            ),
+        ));
+
+        $storageDefinition = new Definition('Symfony\Bridge\Doctrine\HttpFoundation\DbalSessionHandler', array(
+            new Reference('doctrine.dbal.default_connection'),
+        ));
+        $container->setDefinition('session.handler.dbal_handler', $storageDefinition);
+
+You can pass a second parameter to the constructor to set the table name. 
+* ``db_table``: The name of the session table in your database
```

Comment:
you should add an empty before the list. I would also prefer if this was a definition list:

````rst
``db_table``:
    The name of the session table in your database
``db_id_col``
    The name of the id column in your session table (``VARCHAR(255)`` or larger)
``db_data_col``
    The name of the value column in your session table (``TEXT`` or ``GLOB``)
``db_time_col``
    The name of the time column in your session table (``INTEGER``)
````